### PR TITLE
Implemented Verify Page w/ and Set New Login

### PR DESCRIPTION
**Pull request by victoranyun**
### Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Set-up-Staging-DB-f6392fd90a2844cca4514af51df35ae6)

### Implementation description

- Added new page "verify.tsx"
- Set the login page to the latest one
- Created custom function for the login page to capture the email and save it into LocalStorage
- In verify.tsx, use the email to remind the user what email they used

### Steps to test

1. Run locally and try the sign in flow (or on the deployment)
Screen Shot 2021-07-22 at 10 23 45 PM

### What should reviewers focus on?

- Exception handling
- File naming

### Checklist

-   [x] My PR name is descriptive and in imperative tense
-   [x] I have run the linter
-   [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/api/auth/[...nextauth].js [hunk 1]
```diff
@@ -29,6 +29,8 @@ export default NextAuth({
         },
     }),
     pages: {
+        // Redirect to the login.tsx page
+        signIn: "/login",
```
Comment (feng-eric) [inline pages/api/auth/[...nextauth].js:33]: Is this saying on sign in, we want to redirect them to login? Shouldn't we redirect them to home. Let me know if I'm misunderstanding.

File: pages/login.tsx [hunk 2]
```diff
@@ -6,12 +6,23 @@ import {
...
 
 /**
  * This is the page that a user will use to either login or register
  * to the SDC platform
  */
 export default function Login(): JSX.Element {
+    // hook for storing the email
+    const [email, setEmail] = useState("");
+    // save the email into localstorage for email verification page
+    const [, setLocalStorageEmail] = useLocalStorage(
```
Comment (feng-eric) [inline pages/login.tsx:26]: Can you explain the use of local storrage? Is it just for state management?

Comment (feng-eric) [reply pages/login.tsx:26]: If so, for state management, we will use unstated-next. You can find documentation about it in the README

**Review by feng-eric — COMMENTED** [review]
Left some comments

File: pages/login.tsx [hunk 3]
```diff
@@ -6,12 +6,23 @@ import {
...
+        "sdc-email-verification",
+        "",
+    );
+
     return (
         <Center h="500px">
```
Comment (jason17h) [inline pages/login.tsx]: Once https://github.com/uwblueprint/social-diversity-for-children/pull/51 is merged, can you merge staging onto your branch and wrap this page in the `SDCWrapper` component? That'll add the navbar and footer to the page. The PR is small and should be merged sometime today

File: pages/verify.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Box, Center, Text } from "@chakra-ui/react";
+import useLocalStorage from "@utils/useLocalStorage";
+
+export default function Verify(): JSX.Element {
```
Comment (jason17h) [inline pages/verify.tsx:5]: @feng-eric @victoranyun is the goal of this PR to implement the verify page to match the hi-fi design, or to just get something in ASAP for Sunday? If the former, I think there are some minor style changes that might need to be made to match the figma https://www.figma.com/file/ad7RNFErTGP7HkL52edTwQ/SDC-External-Application?node-id=1172%3A15150

File: pages/login.tsx [hunk 5]
```diff
@@ -43,6 +55,10 @@ export default function Login(): JSX.Element {
                         fontSize="10px"
                         fontWeight="400"
                         mt="20px"
+                        onClick={() => {
+                            setLocalStorageEmail(email);
+                            signIn("email", { email: email });
```
Comment (jason17h) [inline pages/login.tsx]: There's a bit of buffer time between when you press the "Continue" button and when it actually redirects - would it be easy to add some sort of loading circle to show that something to show that the button worked and it's just loading? @stacykwok @christinaruu thoughts?

If we go with that, I think you can do something like `const [session, loading] = useSession()` and then if `loading` is true, then you render some sort of loading circle or something like that.

**Comment by jason17h** [comment]
I'm not sure if this is within the scope of this PR, but if I enter an invalid email address (e.g. `asdf`) it redirects me to this `http://localhost:3000/en?error=EmailSignin` which seems right but there isn't a landing page - it's just the homepage. So I'm wondering:
1. Should there be a landing page for when a user passes in an invalid email? (by "invalid" I don't mean non-existent, but I mean something that's not in the format abc@xyz.com) Or should we validate it in the form itself and only allow users to click "Continue" if they've entered a valid email? @christinaruu @stacykwok 
2. Should that be included in this PR or a separate one? thoughts @feng-eric @victoranyun

**Review by jason17h — COMMENTED** [review]
Looks good so far!! I just had some contextual questions + questions for the designers (cc: @stacykwok @christinaruu ) but thanks for implementing this 🤩

File: pages/login.tsx (hunk 3 above)
Comment (jason17h) [reply pages/login.tsx]: Update: it has been merged

File: pages/login.tsx (hunk 5 above)
Comment (christinaruu) [reply pages/login.tsx]: Yess, I think adding a loading screen would be a great idea! Is there a default loading icon or do we need to make/find one?

**Comment by stacykwok** [comment]
> I'm not sure if this is within the scope of this PR, but if I enter an invalid email address (e.g. `asdf`) it redirects me to this `http://localhost:3000/en?error=EmailSignin` which seems right but there isn't a landing page - it's just the homepage. So I'm wondering:
> 
> 1. Should there be a landing page for when a user passes in an invalid email? (by "invalid" I don't mean non-existent, but I mean something that's not in the format [abc@xyz.com](mailto:abc@xyz.com)) Or should we validate it in the form itself and only allow users to click "Continue" if they've entered a valid email? @christinaruu @stacykwok
> 2. Should that be included in this PR or a separate one? thoughts @feng-eric @victoranyun

File: pages/api/auth/[...nextauth].js (hunk 1 above)
Comment (victoranyun) [reply pages/api/auth/[...nextauth].js:33]: This is when we press sign-in on the index page. Not after they sign in. When signIn function is called on the index.tsx

File: pages/verify.tsx (hunk 4 above)
Comment (victoranyun) [reply pages/verify.tsx:5]: @feng-eric I'm not sure about this, please let me know

**Comment by ghost** [comment]
> I'm not sure if this is within the scope of this PR, but if I enter an invalid email address (e.g. `asdf`) it redirects me to this `http://localhost:3000/en?error=EmailSignin` which seems right but there isn't a landing page - it's just the homepage. So I'm wondering:
> 
> 1. Should there be a landing page for when a user passes in an invalid email? (by "invalid" I don't mean non-existent, but I mean something that's not in the format [abc@xyz.com](mailto:abc@xyz.com)) Or should we validate it in the form itself and only allow users to click "Continue" if they've entered a valid email? @christinaruu @stacykwok
> 2. Should that be included in this PR or a separate one? thoughts @feng-eric @victoranyun

For invalid emails, I think they can just be handled by inline validation (i.e. validating it within the form itself and only allowing users to click "continue" if they've entered a valid email). We can draft up screens on handling invalid input + validations

File: pages/login.tsx (hunk 5 above)
Comment (jason17h) [reply pages/login.tsx]: Chakra has some spinners, you can see them here: https://chakra-ui.com/docs/feedback/spinner

Comment (jason17h) [reply pages/login.tsx]: Nvm - it looks like it got added
image

File: pages/login.tsx [hunk 6]
```diff
@@ -5,60 +5,114 @@ import {
...
+                    </Center>
+                    <Center>
+                        <Text fontWeight="400" fontSize="18px" mt="20px">
+                            Registration for Summer 2021 classes begin June 31,
+                            2021
+                        </Text>
+                    </Center>
+                    <FormControl id="email">
+                        <Center>
+                            <Input
```
Comment (jason17h) [inline pages/login.tsx:54]: could you make the font weight a bit lighter?

Comment (victoranyun) [reply pages/login.tsx:54]: which text do you want me to make lighter?

File: pages/verify.tsx (hunk 4 above)
Comment (feng-eric) [reply pages/verify.tsx:5]: Did most of it, just missing the back button now I believe. @jason17h, had a question regarding CSS and the footer btw. PM me when you see this

File: pages/login.tsx (hunk 6 above)
Comment (jason17h) [reply pages/login.tsx:54]: The email text input and also maybe the "Registration for Summer 2021 ..." text above, using the figma design as reference https://www.figma.com/file/ad7RNFErTGP7HkL52edTwQ/SDC-External-Application?node-id=1171%3A15116

image

**Review by feng-eric — APPROVED** [review]
LGTM. We can add the back button and font changes in another PR probably. Can you test one last time before merging?

File: pages/login.tsx (hunk 6 above)
Comment (jason17h) [reply pages/login.tsx:54]: Update - we will fix this in a separate PR

**Review by jason17h — APPROVED** [review]
Looks good, thanks for addressing the comments!

**Pull request merged**